File: pyjs/pyjamas-0.1/pyjs/tests/test015.py
class TimeSlot:

    def __init__(self, dayOfWeek, startMinutes, endMinutes):
        self.dayOfWeek = dayOfWeek
        self.startMinutes = startMinutes
        self.endMinutes = endMinutes
    
    def compareTo(self, other):
        if self.dayOfWeek < other.dayOfWeek:
            return -1
        elif self.dayOfWeek > other.dayOfWeek:
            return 1
        else:
            if self.startMinutes < other.startMinutes:
                return -1
            elif self.startMinutes > other.startMinutes:
                return 1
        return 0

File: pyjs/pyjamas-0.1/pyjs/tests/test015.js
function TimeSlot(dayOfWeek, startMinutes, endMinutes) {
    return new __TimeSlot(dayOfWeek, startMinutes, endMinutes);
}
function __TimeSlot(dayOfWeek, startMinutes, endMinutes) {
    this.dayOfWeek = dayOfWeek;
    this.startMinutes = startMinutes;
    this.endMinutes = endMinutes;
}
__TimeSlot.prototype.compareTo = function(other) {
    if ((this.dayOfWeek < other.dayOfWeek)) {
    return -1;
    }
    else if ((this.dayOfWeek > other.dayOfWeek)) {
    return 1;
    }
    else {
    if ((this.startMinutes < other.startMinutes)) {
    return -1;
    }
    else if ((this.startMinutes > other.startMinutes)) {
    return 1;
    }
    }
    return 0;
};
